### PR TITLE
Enhance commit log

### PR DESCRIPTION
**Pull request by snazy**
Allows to optionally return additional information from the commit log.

The `LogResponse` type has been changed to contain a list of `LogEntry` elements, which contain a `CommitMeta` and, if requested, also the operations (`Put` or `Delete`) of each commit.

The Python CLI behaves as it currently does. The `nessie log` command got a new flag to retrieve and show the additional information.

* Adds new parameter `fetchAdditionalInfo` to commit-log REST endpoint.
* Changes return type of commit-log REST endpoint, returns a list of `LogEntry`
  that contains the `CommitMeta` plus the optionally fetched list of operations.
* Adds a `--extended` flag to Nessie CLI's `log` command to include the additional
  information. Behaves exactly as before without the `--extended` flag.

### CODE REVIEW

File: clients/client/src/main/java/org/projectnessie/client/api/GetCommitLogBuilder.java [hunk 1]
```diff
@@ -30,6 +30,14 @@
     extends QueryBuilder<GetCommitLogBuilder>,
         PagingBuilder<GetCommitLogBuilder>,
         OnReferenceBuilder<GetCommitLogBuilder> {
+
+  /**
+   * Will fetch additional metadata about each commit: operations in a commit and parent hash.
```
Comment (nastra) [inline clients/client/src/main/java/org/projectnessie/client/api/GetCommitLogBuilder.java:35]: maybe "about each commit, such as operations in a commit and parent hash"

File: model/src/test/java/org/projectnessie/api/params/CommitLogParamsTest.java [hunk 2]
```diff
@@ -28,30 +29,53 @@ public void testBuilder() {
...
+  private void verify(
+      Integer maxRecords,
+      String startHash,
+      String endHash,
+      String pageToken,
+      String queryExpression,
+      boolean fetchAdditionalInfo,
+      Supplier<CommitLogParams> generator) {
+    assertThat(generator.get())
+        .isEqualTo(generator.get())
```
Comment (nastra) [inline model/src/test/java/org/projectnessie/api/params/CommitLogParamsTest.java:63]: looks a bit weird to compare `generator.get()` with itself

**Review by nastra — COMMENTED** [review]
LGTM, just a few nits

File: model/src/test/java/org/projectnessie/api/params/CommitLogParamsTest.java (hunk 2 above)
Comment (snazy) [reply model/src/test/java/org/projectnessie/api/params/CommitLogParamsTest.java:63]: It's not the _same_ object

**Review by nastra — APPROVED** [review]

**Pull request merged**